### PR TITLE
Support ElasticSearch 5.0 added

### LINKED ISSUE #149: Support for ElasticSearch 5.0
| Q | A
| --- | ---
| Bug? | no
| New Feature? | yes
| Sulu Version | ^1.5
| Article Version | ^0.3

#### Actual Behavior

Currently only ElasticSearch 2.* is supported

#### Expected Behavior

Also ElasticSearch 5.* support available

### PR DESCRIPTION
**Pull request by kleinkoerkamp**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | yes
| Deprecations? | no
| Fixed tickets | fixes #149 
| License | MIT

#### What's in this PR?

ArticleBundle now also ElasticSearch 5.0 compatible

#### BC Breaks

- Not compatible with PHP 5.5 anymore if you make use of 5.0 bundle

#### To Do

-  Documentation for ONGR/ElasticsearchBundle is different now. Config has changed. I only changed it to the new configuration. Don't if you can see this as Break.

### CODE REVIEW

**Comment by wachterjohannes** [comment]
we should also include that into the test suite:

* one test run for elasticsearch 2.4
* one test run for elasticsearch 5.1

File: Tests/app/AppKernel.php [hunk 1]
```diff
@@ -38,6 +38,11 @@ public function registerContainerConfiguration(LoaderInterface $loader)
             $loader->load(__DIR__ . '/config/versioning.yml');
         }
 
-        $loader->load(__DIR__ . '/config/config.yml');
+        // If version is lower then 5.6 it is an testcase for elasticsearch 2.*, so different ONGR config is needed
+        if (phpversion() < 5.6) {
```
Comment (wachterjohannes) [inline Tests/app/AppKernel.php]: this should be handled via a envorinment variable in travis.

**Comment by wachterjohannes** [comment]
in travis the default elasticsearch version is `1.7` so we have to handle the versions there manually. see https://github.com/wachterjohannes/pucene/blob/b68a2afdb7e95c6bb410b0cb576ae46b29e85319/.travis.yml

File: Tests/app/AppKernel.php (hunk 1 above)
Comment (kleinkoerkamp) [inline Tests/app/AppKernel.php]: @wachterjohannes you're right i will fix this

**Comment by kleinkoerkamp** [comment]
@wachterjohannes hi Johannes, i changed the code for travis, but still get some unknown problems, can you have a look into it? My knowledge of Travis is still a little bit low.

File: Tests/scripts/install-travis-es.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+set -ev
+
+echo $ES_VERSION;
+
+if [[ "${ES_VERSION}" = "2.4.4" ]]; then
```
Comment (wachterjohannes) [inline Tests/scripts/install-travis-es.sh]: i would always use the same installation method (download) this makes the installation script easier to read.

File: .travis.yml [hunk 3]
```diff
@@ -2,25 +2,30 @@ sudo: false
...
+addons:
+  apt:
+    packages:
+      - oracle-java8-set-default
+
 matrix:
   include:
     - php: 5.5
       env:
         - SULU_VERSION="~1.5.1"
```
Comment (wachterjohannes) [inline .travis.yml]: I think if you change this to `~1.5.3` it may work.

File: .travis.yml [hunk 4]
```diff
@@ -59,5 +70,6 @@ script:
   - ./vendor/bin/phpunit $CODE_COVERAGE
 
 after_script:
+  - cat elasticsearch.log
```
Comment (wachterjohannes) [inline .travis.yml]: where have you specified that this file will be written?

File: Resources/doc/installation_es5.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
+# Installation
+
+If you w
```
Comment (wachterjohannes) [inline Resources/doc/installation_es5.md]: there is missing the half sentence (:

File: Tests/app/AppKernel.php [hunk 6]
```diff
@@ -38,6 +38,10 @@ public function registerContainerConfiguration(LoaderInterface $loader)
             $loader->load(__DIR__ . '/config/versioning.yml');
         }
 
-        $loader->load(__DIR__ . '/config/config.yml');
+        if (getenv('ES_VERSION') == '2.4.4') {
```
Comment (wachterjohannes) [inline Tests/app/AppKernel.php]: can you change that by using http://php.net/manual/de/function.version-compare.php ? it should match >=2.2 and <5.0

File: UPGRADE.md [hunk 7]
```diff
@@ -13,3 +13,9 @@ with the cachelifetime resolver.
 Reindex elastic search indexes:
 * bin/adminconsole sulu:article:index-rebuild ###LOCALE### -live
 * bin/adminconsole sulu:article:index-rebuild ###LOCALE###
+
+
+## 0.5.0
+
+Now also support for ElasticSearch 5. To still be compatible with ^2.2, make sure you run:
```
Comment (wachterjohannes) [inline UPGRADE.md]: move that to dev-develop. i will change that when i release the bundle

File: Resources/doc/installation_es5.md (hunk 5 above)
Comment (kleinkoerkamp) [inline Resources/doc/installation_es5.md]: Was not needed anymore.

File: Tests/app/AppKernel.php (hunk 6 above)
Comment (kleinkoerkamp) [inline Tests/app/AppKernel.php]: Fixed,

File: UPGRADE.md (hunk 7 above)
Comment (kleinkoerkamp) [inline UPGRADE.md]: Changed it

File: .travis.yml (hunk 4 above)
Comment (kleinkoerkamp) [inline .travis.yml]: Was not used anymore

**Comment by wachterjohannes** [comment]
@kleinkoerkamp thanks for contribution (:

**Pull request merged**